pkgs/tools/admin/lexicon/default.nix

```diff
@@ -1,60 +1,24 @@
 { lib
-, python3
 , fetchFromGitHub
+, poetry2nix
+, python3
+, pyproject ? ./pyproject.toml
+, poetrylock ? ./poetry.lock
 }:
 
-with python3.pkgs;
+poetry2nix.mkPoetryApplication {
 
-buildPythonApplication rec {
-  pname = "lexicon";
-  version = "3.9.4";
-  format = "pyproject";
+  inherit pyproject poetrylock;
 
-  src = fetchFromGitHub {
+  python = python3;
+
+  projectDir = fetchFromGitHub {
     owner = "AnalogJ";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-TySgIxBEl2RolndAkEN4vCIDKaI48vrh2ocd+CTn7Ow=";
+    repo = "lexicon";
+    rev = "v3.11.2";
```

Comment:
this version information is also available in the toml file. does it make sense to get it from there? this way it also becomes impossible to accidentally check out a rev that does not match the toml file's description.

Reply:
Fixed. It's not necessarily beautiful, though. :)